inc/features/blocks/personalization/edit.js

```diff
@@ -0,0 +1,351 @@
+
+	if ( isLoading ) {
+		return __( 'Loading...', 'altis-experiments' );
+	}
+
+	return '';
+}
+
+// Audience picker input.
+const Edit = ( {
+	addVariant,
+	attributes,
+	className,
+	clientId,
+	copyVariant,
+	isSelected,
+	removeVariant,
+	setAttributes,
+	setVariantAttributes,
+	variants,
+} ) => {
+
+	// Track currently selected variant.
+	const defaultVariantClientId = ( variants.length > 0 && variants[ 0 ].clientId ) || null;
+	const [ activeVariant, setVariant ] = useState( defaultVariantClientId );
+
+	// Track the active variant index to show in the title.
+	const activeVariantIndex = variants.findIndex( variant => variant.clientId === activeVariant );
+
+	// Set clientId attribute if not set.
+	if ( ! attributes.clientId ) {
+		setAttributes( { clientId } );
+	}
+
+	// Ensure variant data is correct parentId is correct.
+	variants.forEach( variant => {
+		if ( ! variant.attributes.parentId || variant.attributes.parentId !== attributes.clientId ) {
+			setVariantAttributes( variant.clientId, {
+				parentId: attributes.clientId,
+			} );
+		}
+	} );
+
+	return (
+		<Fragment>
+			<BlockControls>
+				<Toolbar
+					className="altis-variants-toolbar"
+					controls={ [
+						{
+							icon: 'plus',
+							title: __( 'Add a variant', 'altis-experiments' ),
+							className: 'altis-add-variant-button',
+							onClick: () => {
+								const newClientId = addVariant();
+								setVariant( newClientId );
+							},
+						},
+					] }
+				>
+					{ variants.map( variant => (
+						<Button
+							key={ `variant-select-${ variant.clientId }` }
+							className={ `altis-variant-button components-icon-button has-text ${ activeVariant === variant.clientId && 'is-active' }` }
+							title={ __( 'Select variant', 'altis-experiments' ) }
+							onClick={ () => setVariant( variant.clientId ) }
+						>
+							<VariantTitle variant={ variant } />
+						</Button>
+					) ) }
+				</Toolbar>
+			</BlockControls>
+			<InspectorControls>
+				{ variants.map( variant => {
+					if ( variant.attributes.fallback ) {
+						return (
+							<PanelBody
+								key={ `variant-settings-${ variant.clientId }` }
+								title={ __( 'Fallback', 'altis-experiments' ) }
+							>
+								<p className="description">
+									{ __( 'This variant will be shown as a fallback if no audiences are matched. You can leave the content empty if you do not wish to show anything.', 'altis-experiments' ) }
+								</p>
+							</PanelBody>
+						);
+					}
+
+					return (
+						<PanelBody
+							key={ `variant-settings-${ variant.clientId }` }
+							title={ <VariantTitle variant={ variant } /> }
+						>
+							<AudiencePicker
+								label={ __( 'Audience' ) }
+								audience={ variant.attributes.audience }
+								onSelect={ audience => {
+									setVariantAttributes( variant.clientId, { audience: audience.id } );
+								} }
+								onClearSelection={ () => {
+									setVariantAttributes( variant.clientId, { audience: null } );
+								} }
```

Comment:
Can we make these single lines instead to make it clear they're not complex functions? (Line length will be less of an issue once broken out)